### PR TITLE
[#152543946] Eliminate most of the errors during UAA deploys

### PR DESCRIPTION
**Pull request by bandesz**
## What

Sadly we still see some 503 errors. Out of 6 deploys only 2 had 1-1 errors.

We made the following changes:
 * Nginx was removed from UAA as it was causing some of the downtime (the route-registrar health check was running against UAA and not Nginx) - this removed the 502 errors we saw
 * We enabled the connection draining on the UAA ELB (with default draining timeout which is 300s), which removed the remaining EOF errors but introduced new 503 errors (from the haproxy in front of the gorouter).
 * We set the drain_wait to 120s on the gorouter which decreased the number of 503 errors
 * The uaa.self_service_links_enabled parameter now disables the self-service links (like create account and reset password links), not just removes the links therefore there is no need for nginx anymore.

Minor changes:
 * The uaa.self_service_links_enabled parameter was changed to bool true (was a string value)
 * We had to change the tests checking the create account and reset password links as now UAA handles this and returns with a different status code
 * Reorder+rename forgot password tests
 * Changed the UAA ELB drain timeout from the default 300s to 20s (although it doesn't matter much, but it might help when we remove an instance from an ELB)

Remaining errors:
 * We suspect that haproxy should also drain connections at some point (it doesn't have draining atm). As we timeboxed this story we don't go into it any further. Also there is good chance haproxy will be removed soon anyway which might solve this for good.

Note: we intentionally made separate commits to reflect how the story evolved.

## How to review

1. Deploy from this branch. It is expected that the `api-availability-tests` may fail at this stage.
2. Deploy from a temporary branch that includes the following stemcell change for `router` and `uaa`:
    ```patch
    diff --git a/manifests/cf-manifest/manifest/000-base-cf-deployment.yml b/manifests/cf-manifest/manifest/000-base-cf-deployment.yml
    index 6583319c..dc0dfc39 100644
    --- a/manifests/cf-manifest/manifest/000-base-cf-deployment.yml
    +++ b/manifests/cf-manifest/manifest/000-base-cf-deployment.yml
    @@ -84,6 +84,9 @@ stemcells:
       - alias: default
         name: bosh-aws-xen-hvm-ubuntu-trusty-go_agent
         version: "3468.5"
    +  - alias: test
    +    name: bosh-aws-xen-hvm-ubuntu-trusty-go_agent
    +    version: "3468"

     update:
       canaries: 0
    diff --git a/manifests/cf-manifest/manifest/010-cf-jobs.yml b/manifests/cf-manifest/manifest/010-cf-jobs.yml
    index 0df27b98..ce3fa21a 100644
    --- a/manifests/cf-manifest/manifest/010-cf-jobs.yml
    +++ b/manifests/cf-manifest/manifest/010-cf-jobs.yml
    @@ -141,7 +141,7 @@ jobs:
         vm_type: medium
         vm_extensions:
           - cf_rds_client_sg
    -    stemcell: default
    +    stemcell: test
         networks:
           - name: cf
         properties:
    @@ -329,7 +329,7 @@ jobs:
             release: datadog-for-cloudfoundry
         instances: 2
         vm_type: router
    -    stemcell: default
    +    stemcell: test
         networks:
           - name: router
         properties:
    ```

1. Confirm that the `api-availability-tests` passed or only had some 503 errors (we saw less than 5 errors)

Also check the following:
 * the client IP is still recorded correctly in UAA log

## Who can review

Not @dcarley or @bandesz.

### CODE REVIEW

File: manifests/cf-manifest/manifest/010-cf-jobs.yml [hunk 1]
```diff
@@ -147,43 +145,6 @@ jobs:
...
-          server {
-            listen 9443;
-            ssl on;
-            ssl_certificate_key /var/vcap/jobs/nginx/etc/ssl.key.pem;
-            ssl_certificate  /var/vcap/jobs/nginx/etc/ssl_chained.crt.pem;
-
-            location / {
-              rewrite ^/create_account(.do)?$ /create_account_disabled break;
-              rewrite ^/forgot_password(.do)?$ /create_account_disabled break;
-
```
Comment (henrytk) [inline manifests/cf-manifest/manifest/010-cf-jobs.yml]: How are we now blocking these endpoints?

**Comment by alext** [comment]
The commit message for "Use bool for uaa.login.self_service_links_enabled" links to a non-existent commit in uaa. I suspect you meant to link to uaa-release instead.

**Comment by alext** [comment]
I'm testing this in a dev env now, but after a code review I wonder whether we should make the config for the uaa and cf_router ELBs consistent. Given they both point at the routers it seems odd for them to have different settings for connection draining etc.

**Comment by dcarley** [comment]
> I'm testing this in a dev env now, but after a code review I wonder whether we should make the config for the uaa and cf_router ELBs consistent. Given they both point at the routers it seems odd for them to have different settings for connection draining etc.

I'm not sure. I'm going to create one more follow-up story after the thread on Slack yesterday to improve the app-availability-tests to highlight similar problems with slower or keepalive requests, so perhaps we should tweak it at the same time as that?

**Comment by dcarley** [comment]
> The commit message for "Use bool for uaa.login.self_service_links_enabled" links to a non-existent commit in uaa. I suspect you meant to link to uaa-release instead.

I've rebased it with a correction 👍

**Comment by alext** [comment]
> I'm not sure. I'm going to create one more follow-up story after the thread on Slack yesterday to improve the app-availability-tests to highlight similar problems with slower or keepalive requests, so perhaps we should tweak it at the same time as that?

That makes sense. It's only going to add to the scope of this story otherwise...

**Comment by alext** [comment]
I've successfully run this against dev. Deploying with the temporary stemcell changes resulted in no API downtime reported.

**Pull request merged**